core/src/main/java/bisq/core/offer/OpenOfferManager.java

```diff
@@ -871,39 +871,46 @@ private void maybeUpdatePersistedOffers() {
     ///////////////////////////////////////////////////////////////////////////////////////////
 
     private void republishOffers() {
-        int size = openOffers.size();
-        final ArrayList<OpenOffer> openOffersList = new ArrayList<>(openOffers.getList());
-        if (!stopped) {
-            stopPeriodicRefreshOffersTimer();
-            for (int i = 0; i < size; i++) {
-                // we delay to avoid reaching throttle limits
-
-                long delay = 700;
-                final long minDelay = (i + 1) * delay;
-                final long maxDelay = (i + 2) * delay;
-                final OpenOffer openOffer = openOffersList.get(i);
-                UserThread.runAfterRandomDelay(() -> {
-                    if (openOffers.contains(openOffer)) {
-                        String id = openOffer.getId();
-                        if (id != null && !openOffer.isDeactivated())
-                            republishOffer(openOffer);
-                    }
+        if (stopped) {
+            return;
+        }
 
-                }, minDelay, maxDelay, TimeUnit.MILLISECONDS);
-            }
-        } else {
-            log.debug("We have stopped already. We ignore that republishOffers call.");
+        List<OpenOffer> openOffersList = new ArrayList<>(openOffers.getList());
+        republishOffers(openOffersList);
+
+        stopPeriodicRefreshOffersTimer();
+    }
+
+    private void republishOffers(List<OpenOffer> list) {
+        if (list.isEmpty()) {
+            return;
         }
+
+        OpenOffer openOffer = list.remove(0);
+        if (!openOffers.contains(openOffer) || openOffer.isDeactivated()) {
+            republishOffers(list);
+        }
+
+        republishOffer(openOffer,
+                () -> UserThread.runAfter(() -> republishOffers(list),
+                        30, TimeUnit.MILLISECONDS));
```

Comment:
I don't understand the reason for the delay, all offers will be prepared to republish recursively, then during the recursion unwind they will all get this call so they delay is likely less than 1ms between each offer republish, but all the calls to republish will happen 30ms later. I think this is not much different to republishing without delay.

The previous version had a random delay, but it was a multiple of the number of orders and 700, so 200 orders ran into 140 seconds republish time and each offer would be spaced out in that time.

Reply:
Ah damn, yes you are right...Wanted to avoid the way it was done before, but the delay from the callback is too short to be used as delay. I am acutally not very sure what happens if one publishes 200 offers without delay. It might be that connections to other peers get dropped due dos protection, but not sure if we hit those thresholds. To make it more smooth for the node and the network the delay was used, but could be that it makes things actually worse as it causes more messages (we use bundledEnvelope now to group msg in 1 container).

Maybe best to leave the PR for now open and we do more tests first, just hard to test 200 offers on mainnet ;-(.

Reply:
Yeah, could be it's better to have no delay and get all these messages into one bundle (or just a few) rather than spreading over time